eng/Versions.props

```diff
@@ -22,7 +22,8 @@
     <!--
       Cannot use later versions (5.5.0+) due to https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1302.
     -->
-    <SystemIdentityModelTokensJwtVersion>5.4.0</SystemIdentityModelTokensJwtVersion>
+    <SystemIdentityModelTokensJwtVersion>6.5.1</SystemIdentityModelTokensJwtVersion>
```

Comment:
As per the comment on line 23, this needs reverting as it breaks the Apple provider.

Reply:
Can I just help and fix the Apple provider instead? There seems to be quite a few mentioned of the term "maintaining" in this PR, and yet this legacy dependency is something that can easily be fixed with a little maintenance.

Reply:
I see the issue has been updated since I last looked at it with the fix you've done below, which is acceptable as it doesn't require changes to global configuration that the previous workarounds suggested.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1302#issuecomment-611713183

Fix aside, do you _need_ 6.5.1? Bumping the minimum version for the provider would require consuming applications to go from 5.x to 6.x, which might be a breaking change for them. I think that's also a breaking version in terms of SemVer, and we wouldn't want to have to publish the next release as v4 just for a bump to this package.

Reply:
I just checked and the code for the fix below is in the APIs available to 5.4.0, so unless you **need** 6.x for SuperOffice, revert the dependency (but keep the code fix).

That then allows consumers to use a more recent version if they want to without forcing a dependency update on them and without breaking the Apple provider.

Reply:
I do. I tried 5.4.0, and 5.6.0 and get exception: IDX20803: Unable to obtain configuration from: '[PII is hidden]'. This seems to be related to TLS issues that should be left in the past. 

Unless you know of a work around, I request to bump it up. Otherwise this maybe the deal breaker.

Reply:
If you set `IdentityModelEventSource.ShowPII = true;` in the tests with 5.6.0, what's the real exception message?

If 5.6.0 doesn't work either and it's not something we can resolve, then I suggest we separate the dependency properties so that Apple keeps 5.4.0 as a dependency, and SuperOffice uses the minimum version that works with it that is greater than 5.4.0.

Reply:
Seems to work fine on line 6, then calls the jwks url OK, but fails on 13-14.

1 [xUnit.net 00:00:02.22]         [2020-05-10 16:10:07Z] info: System.Net.Http.HttpClient.Default.LogicalHandler[100]
2 [xUnit.net 00:00:02.22]               Start processing HTTP request POST https://sod.superoffice.com/login/common/oauth/tokens
3 [xUnit.net 00:00:02.22]         [2020-05-10 16:10:07Z] info: System.Net.Http.HttpClient.Default.LogicalHandler[101]
4 [xUnit.net 00:00:02.22]               End processing HTTP request after 0.1035ms - OK
5 [xUnit.net 00:00:02.22]         [2020-05-10 16:10:07Z] info: System.Net.Http.HttpClient.Default.LogicalHandler[100]
6 [xUnit.net 00:00:02.22]               Start processing HTTP request GET https://sod.superoffice.com/login/.well-known/openid-configuration
7 [xUnit.net 00:00:02.22]         [2020-05-10 16:10:07Z] info: System.Net.Http.HttpClient.Default.LogicalHandler[101]
8 [xUnit.net 00:00:02.22]               End processing HTTP request after 0.2062ms - OK
9 [xUnit.net 00:00:02.22]         [2020-05-10 16:10:07Z] info: System.Net.Http.HttpClient.Default.LogicalHandler[100]
10 [xUnit.net 00:00:02.22]               Start processing HTTP request GET https://sod.superoffice.com/login/.well-known/jwks
11 [xUnit.net 00:00:02.22]         [2020-05-10 16:10:07Z] info: System.Net.Http.HttpClient.Default.LogicalHandler[101]
12 [xUnit.net 00:00:02.22]               End processing HTTP request after 0.061ms - OK
13 [xUnit.net 00:00:02.22]         [2020-05-10 16:10:07Z] info: AspNet.Security.OAuth.SuperOffice.SuperOfficeAuthenticationHandler[4]
14 [xUnit.net 00:00:02.22]               Error from RemoteAuthentication: IDX20803: Unable to obtain configuration from: 'https://sod.superoffice.com/login/.well-known/openid-configuration'..
15 [xUnit.net 00:00:02.22]         [2020-05-10 16:10:07Z] info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
16 [xUnit.net 00:00:02.22]               Request finished in 8.2495ms 200 
17 [xUnit.net 00:00:02.24]   Finished:    AspNet.Security.OAuth.Providers.Tests

This works no problem on 6.5.1. I can add a new Version variable for SuperOffice.

Reply:
Versions.prop was updated 4 days ago. It was revert to ensure the Apple provider continues to work and no one should have to update their package.

![image](https://user-images.githubusercontent.com/1012665/81932875-be8db580-95ec-11ea-998e-21828bdaa0a1.png)